File: GitHubPop/app/src/main/java/com/igormeira/githubpop/repository/RepositoryViewHolder.java
package com.igormeira.githubpop.repository;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.igormeira.githubpop.databinding.ItemRepositoryBinding;

public class RepositoryViewHolder extends RecyclerView.ViewHolder {

    ItemRepositoryBinding itemRepositoryBinding;

    public RepositoryViewHolder(@NonNull ItemRepositoryBinding itemRepositoryBinding) {
        super(itemRepositoryBinding.getRoot());
        this.itemRepositoryBinding = itemRepositoryBinding;
    }
}

File: app/src/main/java/app/kelvao/javapop/home/repositorieslist/RepositoryViewHolder.kt
package app.kelvao.javapop.home.repositorieslist

import androidx.recyclerview.widget.RecyclerView.ViewHolder
import app.kelvao.javapop.databinding.HolderHomeRepositoriesBinding
import app.kelvao.javapop.domain.network.response.RepositoryResponse

class RepositoryViewHolder(
    private val binding: HolderHomeRepositoriesBinding,
    private val onClickRepository: ((RepositoryViewHolder) -> Unit)? = null
) : ViewHolder(binding.root) {

    init {
        binding.root.setOnClickListener { onClickRepository?.invoke(this) }
    }

    fun bind(repository: RepositoryResponse) {
        binding.repository = repository
        binding.executePendingBindings()
    }

}